verifiers/internal/gha/builder_test.go

```diff
@@ -486,10 +495,13 @@ func Test_verifyTrustedBuilderID(t *testing.T) {
 			byob:     true,
 		},
 		{
+			// This is a BYOB workflow without an id that tests non-compulsory builder-id
```

Comment:
can we add a unit tests to verify that verification works when no builder ID is provided but it's a trusted builder? Similarly, a test that if the builder is not a trusted one and we don't pass an explicit builder, it fails. These extra tests will need to be in provenance_test.go I think

Reply:
Yes, but, per our discussion, unit tests will have to wait until the release of v1.8.0 of slsa-github-generator. I will make another PR for them after that release is cut.

Reply:
alright so we'll wait until then to merge this PR.

Reply:
actually, we don't need to cut a gen release for those unit tests. provenance.go only verifies the provenance, so you should be able to create those test.

Reply:
The current BYOB Builders [Maven and Gradle (not including nodejs since it has its own separate individual logic)] in v1.7.0 are non-functional in my usage and their readme's on how to use them need to be updated.  I am referring to writing unit tests for VerifyProvenance in `verifiers/internal/gha/provenance_test.go`. Thus, I cannot generate a proper provenance to test `verifyProvenance`. Additionally, I also cannot use provenance generated by builders that are not from a tagged release. If I do not generate the provenance from a tagged release of `slsa-github-generator` then it will fail the first check within the byob conditional of `verifyProvenance`, using `isValidDelegatorBuilderID`. `isValidDelegatorBuilderID` checks for a tagged version via `utils.IsValidBuilderTag`. If it does not get a tagged release an error is thrown.

Once the v1.8.0 release gets cut, the Bazel Builder I added can generate provenance from referencing `slsa-framework/slsa-github-generator@v1.8.0` such that I can use the tagged release provenance for test cases. If the other builders are working, we could add test cases for those as well.

I have the logic for the unit tests ready when the release gets cut; I just need a tagged provenance from a BYOB builder.

Reply:
Test cases added.